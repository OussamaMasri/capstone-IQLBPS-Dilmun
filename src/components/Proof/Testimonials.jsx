import { Trans, useTranslation } from 'react-i18next';
import { people01, people02, people03 } from './assets/FeedbackCardImages';
import FeedbackCard from './FeedbackCard';

const Testimonials = () => {
  const { t } = useTranslation();
  const feedback = [
    {
      id: 'feedback-1',
      content: t('proof.feedback_one_description'),
      name: t('proof.feedback_one_name'),
      title: t('proof.feedback_one_title'),
      img: people01,
    },
    {
      id: 'feedback-2',
      content: t('proof.feedback_two_description'),
      name: t('proof.feedback_two_name'),
      title: t('proof.feedback_two_title'),
      img: people02,
    },
    {
      id: 'feedback-3',
      content: t('proof.feedback_three_description'),
      name: t('proof.feedback_three_name'),
      title: t('proof.feedback_three_title'),
      img: people03,
    },
  ];
  return (
    <section
      id="clients"
      data-testid="testimonials-section"
      className="sm:py-5 py-4 flex justify-center items-center flex-col"
    >
      <div className="w-full flex lg:justify-between justify-center items-center lg:flex-row flex-col sm:mb-4 mb-2">
        <h2
          className="font-bold text-center lg:text-left pl-0 lg:pl-8 xs:text-[48px] text-[40px] text-primary xs:leading-[76.8px] leading-[66.8px] w-full"
          data-testid="testimonials-title"
        >
          <Trans i18nKey="proof.what_they_say" components={{ br: <br /> }} />
        </h2>

          <p
            className="font-poppins text-primary font-normal text-[20px] leading-[30.8px] text-center p-6 lg:text-left md:max-w-[450px]"
            data-testid="testimonials-paragraph"
          >
            {t('proof.everything_you_need')}
          </p>

      </div>

      <div className="container flex flex-wrap sm:justify-center justify-center w-full feedback-container p-6">
        {feedback.map((card) => (
          <FeedbackCard
            key={card.id}
            data-testid="testimonials-card"
            {...card}
          />
        ))}
      </div>
    </section>
  );
};

export default Testimonials;
