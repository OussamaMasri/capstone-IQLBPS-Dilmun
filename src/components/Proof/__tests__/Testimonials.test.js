import "@testing-library/jest-dom";
import { render, screen } from '@testing-library/react';
import Testimonials from '../Testimonials';
import { people01, people02 } from '../assets/FeedbackCardImages';

const MOCK_FEEDBACK = [
  {
    id: 'feedback-1',
    content: 'description',
    name: 'name',
    title: 'title',
    img: people01,
  },
  {
    id: 'feedback-2',
    content: 'description',
    name: 'name',
    title: 'title',
    img: people02,
  },
];
test('should render testimonials component', () => {
    render(<Testimonials feedback={ MOCK_FEEDBACK} />);
    const TestimonialsElement= screen.getByTestId('testimonials-section');
    expect(TestimonialsElement).toMatchSnapshot();
});