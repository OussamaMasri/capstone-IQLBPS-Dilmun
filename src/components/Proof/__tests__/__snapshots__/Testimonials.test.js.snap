// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render testimonials component 1`] = `
<section
  class="sm:py-5 py-4 flex justify-center items-center flex-col"
  data-testid="testimonials-section"
  id="clients"
>
  <div
    class="w-full flex lg:justify-between justify-center items-center lg:flex-row flex-col sm:mb-4 mb-2"
  >
    <h2
      class="font-bold text-center lg:text-left pl-0 lg:pl-8 xs:text-[48px] text-[40px] text-primary xs:leading-[76.8px] leading-[66.8px] w-full"
      data-testid="testimonials-title"
    />
    <p
      class="font-poppins text-primary font-normal text-[20px] leading-[30.8px] text-center p-6 lg:text-left md:max-w-[450px]"
      data-testid="testimonials-paragraph"
    >
      proof.everything_you_need
    </p>
  </div>
  <div
    class="container flex flex-wrap sm:justify-center justify-center w-full feedback-container p-6"
  >
    <div
      class="flex flex-col px-5 py-6 rounded-[20px] max-w-[370px] md:mr-10 sm:mr-5 mr-0 my-5 hover:bg-tertiary ease-in-out duration-500"
      data-testid="feedbackcard-div"
    >
      <img
        alt="double_quotes"
        class="w-[42.6px] h-[27.6px] object-contain"
        src="quotes.svg"
      />
      <p
        class="font-poppins font-normal text-[17px] leading-[32px] text-primary my-10"
        data-testid="feedback-card-content"
      >
        proof.feedback_one_description
      </p>
      <div
        class="flex flex-row"
      >
        <img
          alt="proof.feedback_one_name"
          class="w-[48px] h-[48px] rounded-full"
          src="people01.png"
        />
        <div
          class="flex flex-col ml-4 hover:text-background text-primary"
        >
          <h4
            class="font-poppins font-semibold text-[19px] leading-[32px] "
            data-testid="feedback-card-name"
          >
            proof.feedback_one_name
          </h4>
          <p
            class="font-poppins font-normal text-[15px] leading-[24px] text-left"
            data-testid="feedback-card-title"
          >
            proof.feedback_one_title
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col px-5 py-6 rounded-[20px] max-w-[370px] md:mr-10 sm:mr-5 mr-0 my-5 hover:bg-tertiary ease-in-out duration-500"
      data-testid="feedbackcard-div"
    >
      <img
        alt="double_quotes"
        class="w-[42.6px] h-[27.6px] object-contain"
        src="quotes.svg"
      />
      <p
        class="font-poppins font-normal text-[17px] leading-[32px] text-primary my-10"
        data-testid="feedback-card-content"
      >
        proof.feedback_two_description
      </p>
      <div
        class="flex flex-row"
      >
        <img
          alt="proof.feedback_two_name"
          class="w-[48px] h-[48px] rounded-full"
          src="people02.png"
        />
        <div
          class="flex flex-col ml-4 hover:text-background text-primary"
        >
          <h4
            class="font-poppins font-semibold text-[19px] leading-[32px] "
            data-testid="feedback-card-name"
          >
            proof.feedback_two_name
          </h4>
          <p
            class="font-poppins font-normal text-[15px] leading-[24px] text-left"
            data-testid="feedback-card-title"
          >
            proof.feedback_two_title
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col px-5 py-6 rounded-[20px] max-w-[370px] md:mr-10 sm:mr-5 mr-0 my-5 hover:bg-tertiary ease-in-out duration-500"
      data-testid="feedbackcard-div"
    >
      <img
        alt="double_quotes"
        class="w-[42.6px] h-[27.6px] object-contain"
        src="quotes.svg"
      />
      <p
        class="font-poppins font-normal text-[17px] leading-[32px] text-primary my-10"
        data-testid="feedback-card-content"
      >
        proof.feedback_three_description
      </p>
      <div
        class="flex flex-row"
      >
        <img
          alt="proof.feedback_three_name"
          class="w-[48px] h-[48px] rounded-full"
          src="people03.png"
        />
        <div
          class="flex flex-col ml-4 hover:text-background text-primary"
        >
          <h4
            class="font-poppins font-semibold text-[19px] leading-[32px] "
            data-testid="feedback-card-name"
          >
            proof.feedback_three_name
          </h4>
          <p
            class="font-poppins font-normal text-[15px] leading-[24px] text-left"
            data-testid="feedback-card-title"
          >
            proof.feedback_three_title
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
